extensions/rich_text_components/Video/Video.js

```diff
@@ -34,6 +34,8 @@ oppia.directive('oppiaNoninteractiveVideo', [
 
           $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
             $attrs.videoIdWithValue);
+          $scope.videoId = $scope.videoId.match(
+            /([A-z]|[0-9]){11}/g);
```

Comment:
I think your regex is missing some valid characters. The length of a youtube video is 11 characters and there's 64 possibilities for every character.
- A-Z (26)
- a-z (26)
- - _(2)
See [here](http://stackoverflow.com/questions/6180138/whats-the-maximum-length-of-a-youtube-video-id) for more info

Also, you need to add the case-insensitive flag if you want to go with your current regex or you can make use of `\w` which captures alphabetic characters, numerals and underscores. You might want to take care of hypens though.

And match function always returns an array of the entire match and matches in captured groups if any.
For ref, read [this](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/match)